src/libponyc/type/cap.c

```diff
@@ -121,11 +101,8 @@ bool is_cap_sub_cap(token_id sub, token_id subalias, token_id super,
         case TK_REF:
         case TK_VAL:
         case TK_BOX:
-        case TK_CAP_READ:
```

Comment:
@sylvanc Note that I removed this case here:

1. `#read == {iso, trn, ref, val, box}`
2. and `trn` is not a subtype of `iso`
3. therefore, using the rule that "every possible instantiation of sub must be a subtype of every possible instantiation of super", `trn` is not a subtype of `#read`

Right?